### PR TITLE
[List] Update snapshot golden images for dynamic type change.

### PR DESCRIPTION
**Pull request by fluxminds**
Golden images didn't get updated when List dynamic type pull request was merged.

### CODE REVIEW

**Review by romoore — DISMISSED** [review]
Looks like there's a layout issue in the ExtraLarge image. Can you please fix the test to relayout correctly?  I will also help to set the backgroundColor on the List Item view so we can see where it ends and the background begins.

**Comment by fluxminds** [comment]
@romoore I think the layout is correct (since we don't do resizing cell itself), there might be a layout issue in cell implementation itself. I can force a relayout in 4937cfb2c7044bed24093f75c470cf8fcceb5501, (a layout was triggered correctly before snapshot verification before).

**Review by jverkoey — APPROVED** [review]

**Review by romoore — DISMISSED** [review]
Looks like the ExtraLarge is still broken - is Lists broken?

**Comment by fluxminds** [comment]
@romoore I commented above on what is happening. List doesn't support sizeToFit, and its size is  supposed to be adjusted in Autolayout. The golden images here are expected.

**Comment by romoore** [comment]
@wenyuzhang666 If the List view expects Autolayout to work (at all), the Snapshot tests should use AutoLayout.

**Review by romoore — CHANGES_REQUESTED** [review]
Please open an Issue to track the bug in these Snapshot tests or else please correct the Snapshot tests.

**Comment by jverkoey** [comment]
Out of curiousity how did we land the previous change without updating the snapshot tests?

**Comment by romoore** [comment]
@jverkoey We aren't running presubmits against an integrated/merged version of the PR branch.

https://github.com/material-components/material-components-ios/issues/6280

**Comment by fluxminds** [comment]
@jverkoey I think the later one that did the functional change passed CI before develop branch got updated. Our CI doesn't rerun if a test depending on it is changed on develop branch.

File: components/List/tests/snapshot/MDCSelfSizingStereoCellSnapshotTests.m [hunk 1]
```diff
@@ -97,6 +97,8 @@ - (void)generateSnapshotAndVerifyForView:(UIView *)view {
   [self snapshotVerifyView:snapshotView];
 }
 
+// TODO(#7487): The size of the cell view sent for snapshot is not correct because
```
Comment (romoore) [inline components/List/tests/snapshot/MDCSelfSizingStereoCellSnapshotTests.m]: Nit: Please use the URL of the Issue, not just the number.

**Review by romoore — APPROVED** [review]
Approving with the understanding that @wenyuzhang666 will be fixing these tests in an immediate follow-up PR.

File: components/List/tests/snapshot/MDCSelfSizingStereoCellSnapshotTests.m (hunk 1 above)
Comment (fluxminds) [reply components/List/tests/snapshot/MDCSelfSizingStereoCellSnapshotTests.m]: Addressed.

**Comment by fluxminds** [comment]
There probably will be not an immediate follow up PR since I don't feel it is blocking verifying dynamic type (text size change) at this stage.

**Review by romoore — DISMISSED** [review]
In that case, please correct the test here.  If we aren't going to correct it as a follow-up, we shouldn't knowing commit new broken code like this. This test is very new and should not be "left" as broken.

**Comment by fluxminds** [comment]
This test still has values for verifying the dynamic type feature. It is not perfect, but not broken. I disagree that this code shouldn't be committed when TODO follow up is added.

**Comment by jverkoey** [comment]
I'm not sure the specifics here, but please do the following in this order of priority:

1. Unbreak `develop`.
2. Fix these tests to their correct / desired behavior.

**Pull request merged**